src/app/report/report-parameter/_shared/chip-list-fieldset/chip-list-fieldset.component.ts

```diff
@@ -0,0 +1,138 @@
+import {Component, Input, OnDestroy, OnInit, ViewChild} from '@angular/core';
+import {FormControl, ValidatorFn} from "@angular/forms";
+import {COMMA, ENTER, SPACE} from "@angular/cdk/keycodes";
+import {ReportType} from 'shared/model';
+import {MatChipInputEvent, MatChipList} from "@angular/material/chips";
+import {Subscription} from "rxjs";
+import {ReportService} from "../../../report.service";
+
+@Component({
+	selector: 'ec-chip-list-fieldset',
+	templateUrl: './chip-list-fieldset.component.html',
+	styleUrls: ['./chip-list-fieldset.component.scss']
+})
+export class ChipListFieldsetComponent implements OnInit, OnDestroy {
+
+	@Input()
+	chipDataFormControl: FormControl
+
+	@Input()
+	formatValidator: ValidatorFn
+
+	@ViewChild('chipList')
+	chipList: MatChipList;
+
+	ReportType = ReportType;
+	separatorKeysCodes: number[] = [ENTER, COMMA, SPACE];
+
+	// Used for currently entered data for a chip to be added.
+	formControl = new FormControl('', {updateOn: 'submit'});
+	errorUvcis: string[] = [];
+
+	subscription: Subscription;
+
+	constructor(private readonly reportService: ReportService) {
+	}
+
+	ngOnInit(): void {
+		this.subscription = this.reportService.reset$.subscribe(() => this.resetInput())
+
+		// Trigger new change detection run because updateErrorState() changes formGroup validity
+		// which is used in a component higher in the tree resulting in ExpressionChangedAfterItHasBeenCheckedError.
+		setTimeout(() => this.updateErrorState());
+	}
+
+	ngOnDestroy() {
+		this.subscription?.unsubscribe()
+	}
+
+	onPaste(event: ClipboardEvent): void {
+		event.preventDefault(); //Prevents the default action
+		event.clipboardData
+			.getData('Text')
+			// Accepts:
+			// xxx,xxx,xxx
+			// xxx, xxx, xxx
+			// xxx xxx xxx
+			// xxx\nxxx\nxxx
+			.split(/, |,| |\n/)
```

Comment:
Why limit the format to a single space? Imho "xxx   xxx       xxxx" and "xxx,  xxx,      " should be valid aswell.

Reply:
`"xxx xxx xxxx"` and `"xxx, xxx, "` are is valid. E.g. `"xxx xxx xxxx".split(/, |,| |\n/)` returns ["xxx","xxx","xxx"] which is correct. Or I don't understand the question. :)